servers/src/main/java/tachyon/worker/block/allocator/GreedyAllocator.java

```diff
@@ -27,7 +28,7 @@
  * A greedy allocator that returns the first Storage dir fitting the size of block to allocate.
  * This class serves as an example how to implement an allocator.
  */
-public class GreedyAllocator implements Allocator {
+public class GreedyAllocator extends BlockStoreEventListenerBase implements Allocator {
```

Comment:
@calvinjia you mean make Evictor/Allocator extend BlockStoreEventListener? I don't think they (interface) can extend BlockStoreEventListenerBase (class). In this case each implementation of Evitor/Allocator will have to implement all the methods specified by BlockStoreEventListener, right? Is that what we want? Shall we introduce an abstract class here?